earthfile2llb/wait_block.go

```diff
@@ -198,16 +199,19 @@ func (wb *waitBlock) saveImages(ctx context.Context) error {
 				if err != nil {
 					return err
 				}
-				if item.c.opt.UseLocalRegistry {
-					localRegPullID := pullPingMap.Insert(sessionID, platformImgName)
-					gwCrafter.AddMeta(fmt.Sprintf("%s/export-image-local-registry", refPrefix), []byte(localRegPullID))
-				} else {
-					gwCrafter.AddMeta(fmt.Sprintf("%s/export-image", refPrefix), []byte("true"))
+				if !item.c.opt.UseLocalRegistry {
+					return errors.Errorf(
+						"the --uselocalregistry feature must also be enabled in order to use multi-platform images in combination with the --wait-block feature")
```

Comment:
Note that in the context of satellites, we still use the old tar-based exporter... I think we ignore this flag in that case.

Should we support tar-based exporting here too for the satellite case?

Reply:
interesting point. I was using the pullping map to "smuggle" the manifest from this code to the builder code -- i'll have to do some testing against a satellite to see what breaks.

Reply:
Let me know if you need to pair up - I'm in and out this week with an unpredictable schedule, but I can make myself available if you need me.

Reply:
I had to pass along a reference to the pullpingmap, along with a list of manifest keys (in order to know when all images from the wait block have been exported); it's a bit ugly since the pullpingmap is used for both onPull and onImage now.

maybe it should be renamed to manifestSmuggler, or something more generic?

if you have other ideas on how to achieve this, i'm keen to pair up.

Reply:
Do I understand correctly that this is a temporary hack until we move off of the old builder code?

Reply:
unfortunately not all of it -- the `onImage` and `onImageDone` functions are required for the new code.

Reply:
Makes sense - this could probably be improved, but it could also be a future PR.

Perhaps the PullPingMap needs to evolve into something more complex, or something.